### ISSUE TITLE
cmake version is not used when the library is built

### ISSUE THREAD

**Issue by GiulioRomualdi** [robotology/yarp-telemetry#164]
I noticed that even if the version is set in the `CMakeLists.txt`, the generated libraries uses the version of the yarp installation found in the computer

```
-- Install configuration: "Release"
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/cmake/YARP_telemetry/YARP_telemetryConfigVersion.cmake
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/cmake/YARP_telemetry/YARP_telemetryConfig.cmake
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/cmake/YARP_telemetry/YARP_telemetryTargets.cmake
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/cmake/YARP_telemetry/YARP_telemetryTargets-release.cmake
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/libYARP_telemetry.so.3.6.0
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/libYARP_telemetry.so.0
-- Set runtime path of "/home/gromualdi/robot-code/robotology-superbuild/build/install/lib/libYARP_telemetry.so.3.6.0" to "$ORIGIN/:$ORIGIN/../lib:$ORIGIN/../:/home/gromualdi/robot-code/robotology-superbuild/build/install/lib"
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/libYARP_telemetry.so
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/api.h
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/Buffer.h
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/BufferConfig.h
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/BufferManager.h
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/Record.h
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/include/yarp/telemetry/experimental/TreeNode.h
-- Up-to-date: /home/gromualdi/robot-code/robotology-superbuild/build/install/share/yarp/plugins/yarp-telemetry.ini
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/lib/yarp/yarp_telemetryDeviceDumper.so
-- Set runtime path of "/home/gromualdi/robot-code/robotology-superbuild/build/install/lib/yarp/yarp_telemetryDeviceDumper.so" to "$ORIGIN/:$ORIGIN/../lib:$ORIGIN/../:/home/gromualdi/robot-code/robotology-superbuild/build/install/lib"
-- Installing: /home/gromualdi/robot-code/robotology-superbuild/build/install/share/yarp/plugins/telemetryDeviceDumper.ini
```

This is the content of `YARP_telemetryConfigVersion.cmake`

```cmake
# This is a basic version file for the Config-mode of find_package().
# It is used by write_basic_package_version_file() as input file for configure_file()
# to create a version-file which can be installed along a config.cmake file.
#
# The created file sets PACKAGE_VERSION_EXACT if the current version string and
# the requested version string are exactly the same and it sets
# PACKAGE_VERSION_COMPATIBLE if the current version is >= requested version,
# but only if the requested major version is the same as the current one.
# The variable CVF_VERSION must be set before calling configure_file().


set(PACKAGE_VERSION "3.6.0+7-20220107.2+git38581de70")

if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION)
  set(PACKAGE_VERSION_COMPATIBLE FALSE)
else()

  if("3.6.0+7-20220107.2+git38581de70" MATCHES "^([0-9]+)\\.")
    set(CVF_VERSION_MAJOR "${CMAKE_MATCH_1}")
  else()
    set(CVF_VERSION_MAJOR "3.6.0+7-20220107.2+git38581de70")
  endif()

  if(PACKAGE_FIND_VERSION_MAJOR STREQUAL CVF_VERSION_MAJOR)
    set(PACKAGE_VERSION_COMPATIBLE TRUE)
  else()
    set(PACKAGE_VERSION_COMPATIBLE FALSE)
  endif()

  if(PACKAGE_FIND_VERSION STREQUAL PACKAGE_VERSION)
      set(PACKAGE_VERSION_EXACT TRUE)
  endif()
endif()


# if the installed project requested no architecture check, don't perform the check
if("FALSE")
  return()
endif()

# if the installed or the using project don't have CMAKE_SIZEOF_VOID_P set, ignore it:
if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR "8" STREQUAL "")
  return()
endif()

# check that the installed version has the same 32/64bit-ness as the one which is currently searching:
if(NOT CMAKE_SIZEOF_VOID_P STREQUAL "8")
  math(EXPR installedBits "8 * 8")
  set(PACKAGE_VERSION "${PACKAGE_VERSION} (${installedBits}bit)")
  set(PACKAGE_VERSION_UNSUITABLE TRUE)
endif()
```

cc @traversaro @Nicogene @S-Dafarra @AlexAntn

**Comment by GiulioRomualdi** [comment]
This Patch fixes the problem. However I don't know if this was done intentionally

```patch
diff --git a/cmake/YarpInstallBasicPackageFiles.cmake b/cmake/YarpInstallBasicPackageFiles.cmake
index 87daa21..5a349db 100644
--- a/cmake/YarpInstallBasicPackageFiles.cmake
+++ b/cmake/YarpInstallBasicPackageFiles.cmake
@@ -56,7 +56,7 @@ macro(YARP_INSTALL_BASIC_PACKAGE_FILES _export)
   endif()
   install_basic_package_files(${_export}
                               ${_config_template}
-                              VERSION ${YARP_VERSION}
+                              VERSION ${yarp-telemetry_VERSION}
                               COMPATIBILITY SameMajorVersion
                               EXPORT_DESTINATION "${CMAKE_BINARY_DIR}/${_export}"
                               INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${_export}"
diff --git a/src/libYARP_telemetry/src/CMakeLists.txt b/src/libYARP_telemetry/src/CMakeLists.txt
index 0fcc872..34e9b83 100644
--- a/src/libYARP_telemetry/src/CMakeLists.txt
+++ b/src/libYARP_telemetry/src/CMakeLists.txt
@@ -69,8 +69,7 @@ list(APPEND YARP_telemetry_PRIVATE_DEPS nlohmann_json)
 
 set_property(TARGET YARP_telemetry PROPERTY PUBLIC_HEADER ${YARP_telemetry_HDRS})
 set_property(TARGET YARP_telemetry PROPERTY PRIVATE_HEADER ${YARP_telemetry_IMPL_HDRS})
-set_property(TARGET YARP_telemetry PROPERTY VERSION ${YARP_VERSION_SHORT})
-set_property(TARGET YARP_telemetry PROPERTY SOVERSION 0)
+set_property(TARGET YARP_telemetry PROPERTY VERSION ${yarp-telemetry_VERSION})
 set_property(TARGET YARP_telemetry PROPERTY FOLDER "Libraries")
 
 install(
```

**Comment by Nicogene** [comment]
I don't know if this is something intended, since it is designed as a YARP component, maybe it has to have the same version ? @drdanz @traversaro

**Comment by Nicogene** [comment]
This is fixed by #171
